### ISSUE TITLE
Transparency effect does not work on Fedora GNOME

### ISSUE THREAD

**Issue by MarcusWichelmann** [AvaloniaUI/Avalonia#5333]
**Describe the bug**
When the `TransparencyLevelHint` property of a window is set to e.g. `AcrylicBlur` and the window background is (partially-)transparent (`Background="#C0000000"` or `Background="{x:Null}"`), the transparency effect is not visible on Fedora with GNOME.

@Splitwirez Ran the very same binaries on his linux system and the transparency was visible there. So it has probably something to do with my distribution or GNOME.

Gitter.im chat history of us trying to track this down: [22. Januar 2021 22:21](https://gitter.im/AvaloniaUI/Avalonia?at=600b41d3cf8b8277344a0eff)

**To Reproduce**
1. Create a new application using the `avalonia.app` dotnet-new template
2. Edit `MainWindow.axaml` and add `TransparencyLevelHint="AcrylicBlur" Background="#C0000000"` to the window.
3. `dotnet run`
4. A black window is shown with no transparency.

I'm able to reproduce this on two different machines (one with NVIDIA graphics, one with Intel graphics) with X11 and Wayland.

**Expected behavior**
- A (partially) transparent window pops up as it does on other systems.
- In cases, where transparency is not supported, I would expect a noisy gray background, like it's also visible in the Controls Gallery app.

**Screenshots**
Test application:
![image](https://user-images.githubusercontent.com/5501487/105555071-ec450580-5d08-11eb-8ee3-36abe7a06ad9.png)

Controls Gallery (no transparency too, but gray background):
![image](https://user-images.githubusercontent.com/5501487/105555650-282c9a80-5d0a-11eb-98e4-df16261dedba.png)

**Desktop:**
 - OS: Linux, Fedora 33, GNOME Desktop
 - Desktop Environment: GNOME 3.38.2
 - Version: 0.10.0 stable release

**Minimal example project**
Here is the minimal project that I used to reproduce the issue as well as the binaries: https://drive.wichelmann.cloud/s/3JYtWjEbCzmbin3

**Comment by kekekeks** [comment]
Unfortunately GNOME Shell doesn't support blur-behind effect, only regular transparency. 

KDE supports blur-behind, but Blur level is not strong enough to be considered "acrylic" so it's reported as TransparencyLevel.Blur.

Consider filing an issue in GNOME shell to either add support for `_KDE_NET_WM_BLUR_BEHIND_REGION` atom or to add another way to apply blurbehind effect.

**Comment by kekekeks** [comment]
@danwalmsley any idea why transparency might be completely disabled by the fluent theme? Window Background should still be applied properly

**Comment by kekekeks** [comment]
@MarcusWichelmann I've run your project on KDE and observe the following effect:
![изображение](https://user-images.githubusercontent.com/1067584/105557603-aba3b700-5d1d-11eb-8eab-af37e886404c.png)

I've changed the hint to `Transparent` and see:
![изображение](https://user-images.githubusercontent.com/1067584/105557689-df7edc80-5d1d-11eb-82f9-752d72e92028.png)


So it's either yet another GNOME quirk or our X11 visual selection has failed.

Could you please run `glxinfo`? Also please check if it's a Wayland or X11 session

**Comment by MarcusWichelmann** [comment]
@kekekeks Ah, thanks. I think that's fine if the blur effect doesn't work in this environment. But the transperancy should, right?

Also, where is this noisy-gray background of the XamlControlsGallery coming from? I'm wondering why it isn't there in the `avalonia.app` template. IMO this would be a way better fallback/default look, especially when transparency does not work. Currently, the window is just black/white, and the design of the fluent controls on top looks very out of place that way.

**Comment by MarcusWichelmann** [comment]
@kekekeks As mentioned above, it's occuring with X11 and Wayland.

glxinfo: https://pastebin.com/8yCuC07a

**Comment by danwalmsley** [comment]
Noisy background is `<ExperimentalAcrylicBorder>` control, see 
https://github.com/AvaloniaUI/XamlControlsGallery/blob/master/XamlControlsGallery/Views/MainView.xaml#L14

for example usage.

**Comment by kekekeks** [comment]
Mkay, reproduced with nvidia + bumblebee

**Comment by MarcusWichelmann** [comment]
@danwalmsley Ah, thank you. Have checked the code, but have not seen that. Very nice.

**Comment by danwalmsley** [comment]
@MarcusWichelmann so you know, this is a stop gap until our new composition engine is ready. Which will allow acrylic to be supported properly and in-app acrylic too.

**Comment by MarcusWichelmann** [comment]
Aah. I suppose that's why it's not the default in the template right now? Makes sense.

**Comment by danwalmsley** [comment]
Yes, and as you can see, only OSX and Windows support it properly.

**Comment by kekekeks** [comment]
@MarcusWichelmann nevermind, did NOT reproduce the issue, it failed to become transparent because of `primusrun`. With `optirun` and actual NVIDIA-driven configuration everything is OK.

Please try to compile this program http://paste.ubuntu.com/p/QrtHTBpmkn/ (`gcc cube.c  -lGLX -lX11 -lGL -lm -lXrender -o cube`) and see if transparency works on your system there.

**Comment by kekekeks** [comment]
@MarcusWichelmann please try setting TransparencyHint to Transparent. It seems that instead of falling back to regular transparency our fluent theme just places a huge black rectangle on the window background.

**Comment by MarcusWichelmann** [comment]
@kekekeks 

> Please try to compile this program http://paste.ubuntu.com/p/QrtHTBpmkn/ (gcc cube.c -lGLX -lX11 -lGL -lm -lXrender -o cube) and see if transparency works on your system there.

It does.
![image](https://user-images.githubusercontent.com/5501487/105576481-45498380-5d73-11eb-836a-0ab4fff90a02.png)

> please try setting TransparencyHint to Transparent.

Ah, that works indeed. Now the window is transparent. So it's only making the wrong decision for the fallback?

**Comment by kekekeks** [comment]
>So it's only making the wrong decision for the fallback?

Yes, which is either a bug or API design issue

**Comment by ArchitektApx** [comment]
I can confirm pretty much the same problem on:
Manjaro Linux 21.0.2
Gnome v40 with GDM-PRIME/Optimus on X11
AvaloniaUI 0.10.2

ControlCatalog:
integrated: no transparency whatsoever
hybrid/nvidia: same

`Transparent` setting in xaml:
integrated: works, background is seethrough
hybrid/nvidia: same

cube demo in here:
integrated: background not visible
hybrid/nvidia: full black no transparency
